app/puppet-access/login.go

```diff
@@ -0,0 +1,58 @@
+package app
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/puppetlabs/pe-sdk-go/api/puppet-access/client/operations"
+	"github.com/puppetlabs/pe-sdk-go/api/puppet-access/models"
+	"github.com/puppetlabs/pe-sdk-go/token/filetoken"
+)
+
+func (puppetAccess *PuppetAccess) getToken() (string, error) {
+	client, err := puppetAccess.Client.GetClient()
+	if err != nil {
+		return "", err
+	}
+
+	loginParameters := operations.NewLoginParamsWithContext(context.Background())
+	body := models.LoginRequest{
+		Login:    puppetAccess.Username,
+		Password: puppetAccess.Password,
+		Lifetime: puppetAccess.Lifetime,
+		Label:    puppetAccess.Label,
+	}
+	loginParameters.SetBody(&body)
+	response, err := client.Operations.Login(loginParameters)
+
+	if err != nil {
+		return "", err
+	}
+
+	if response.Payload.Token == "" {
+		return "", fmt.Errorf("The response did not contain a token. Rerun with --debug to see full body")
+	}
+
+	return response.Payload.Token, nil
+}
+
+func (puppetAccess *PuppetAccess) saveToken(token string) error {
+	filetoken := filetoken.NewFileToken(puppetAccess.TokenPath)
+	return filetoken.Write(token)
+}
+
+// Login method requests a token from the server
+func (puppetAccess *PuppetAccess) Login(print bool) error {
+
+	token, err := puppetAccess.getToken()
+	if err != nil {
+		return err
+	}
+
+	if print {
+		fmt.Println(token)
+		return nil
+	}
+
+	return puppetAccess.saveToken(token)
```

Comment:
Maybe there is a better solution than passing the print parameter to the Login method. Not sure if the printing should be handled by the SDK or the CLI

Reply:
Login could always return (string, error) and could be cli job to show it or not

Reply:
I fact I read documentation and we still need the flag, since we should not save it if `print` is true

Reply:
If we would handle this on CLI, can't we call the print/save action there?

Reply:
I also think it's better that the output should not be handled by the SDK.
Login method will still receive the print parameter just so it does not save the token to disk.

Reply:
Yes, I think it is better that app/* is not directly calling token/*